metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/NetworkFunctions.java

```diff
@@ -232,10 +237,15 @@ private static URL toUrl(Object urlObj) {
     if(urlObj == null) {
       return null;
     }
-    try {
-      return new URL(urlObj.toString());
-    } catch (MalformedURLException e) {
-      return null;
+    if(urlObj instanceof String) {
+      try {
+        return new URL(urlObj.toString());
+      } catch (MalformedURLException e) {
```

Comment:
Why would Malformed Url not be an exception as well?   It would be much better to get a detailed error than null.